pydnd/equipment/views.py

```diff
@@ -170,17 +172,81 @@ def get(self, request, name_or_id):
         return Response(queryset_dict, status=status.HTTP_200_OK)
 
 
-class ArmorList(APIView):
+class WeaponList(APIView):
 
     def get(self, request):
-        armor = Armor.objects.all()
-        data = ArmorListSerializer(armor, many=True).data
+        weapon = Weapon.objects.all()
+        data = WeaponListSerializer(weapon, many=True).data
         return Response(data)
 
 
-class WeaponList(APIView):
+#TODO Remove Post
+#List of equipment sub categories
+class ArmorList(generics.ListCreateAPIView):
 
-    def get(self, request):
-        weapon = Weapon.objects.all()
-        data = WeaponListSerializer(weapon, many=True).data
-        return Response(data)
\ No newline at end of file
+    def post(self, request, *args, **kwargs):
+
+        data = request.data
+
+        armor_category = get_attribute_by_name(data, 'armor_category', ArmorCategory)
+
+        armor = ArmorSerializer(data=data)
+        if armor.is_valid():
+            armor_object = armor.save()
+            armor_object.armor_category = armor_category
+            armor_object = armor_object.save()
+        else:
+            return Response(status=status.HTTP_400_BAD_REQUEST)
+        return Response(data)
```

Comment:
Probably want to return `model_to_dict(armor_object)`. You probably looked at the example i have where I'm not doing that. Practice what you preach pete.